### PR TITLE
serialize errors using eth-json-rpc-errors; add .nvmrc

### PR DESCRIPTION
**Pull request by rekmarks**
Refactor `serializeError` function to use implementation in new module, [`eth-json-rpc-errors`](https://github.com/MetaMask/eth-json-rpc-errors).

The implementation begins here: https://github.com/MetaMask/eth-json-rpc-errors/blob/master/index.js#L26

The `JsonRpcError` interface now extends `IJsonRpcError`, located here: https://github.com/MetaMask/eth-json-rpc-errors/blob/master/index.d.ts#L2-L7

Additionally, PR includes minor fixes such as removed unused functions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "@babel/preset-env": "^7.3.4",
     "async": "^2.0.1",
     "babelify": "^10.0.0",
+    "eth-json-rpc-errors": "github:MetaMask/eth-json-rpc-errors",
```
Comment (danfinlay) [inline package.json]: Let's get this published to npm, and added to the MetaMask npm group (will allow npm/yarn to do its appropriate hash-locking).

**Review by danfinlay — DISMISSED** [review]
Needs to include a breaking version change in case downstream dependents are relying on current (crappy) error behavior.

I'd also like to make sure @kumavis gets a sign-off on this at least so he's aware of the change.

For convenience, here is a link to [eth-json-rpc-errors](https://github.com/metamask/eth-json-rpc-errors).

File: package.json (hunk 1 above)
Comment (rekmarks) [reply package.json]: Pushed new commit with [eth-json-rpc-errors](https://www.npmjs.com/package/eth-json-rpc-errors) installed from npm. I'll leave the adding to the MM org to you.

**Review by danfinlay — APPROVED** [review]
Looks good to me now.

**Pull request merged**